File: Python/Easy/twosum.py
nums = [1, 2, 3, 4, 7]
target = 10
def twoSum(list, target):
  # brute force solution
  # for i in range(len(nums)):
  #   for j in range(i+1, len(nums)):
  #     if nums[i] + nums[j] == target:
  #       return [i, j]
  
  #optimize solution using hashmap
  hash_map = dict()
  for index, value in enumerate(list):
    rem = target - value
    if rem in hash_map:
      return [hash_map[rem], index]
    hash_map[value] = index

print(twoSum(nums, target))

File: Js/Easy/TwoSum.js
//Brute force solution
function twoSum(nums, target) {
  var l = nums.length;
  for (i = 0; i < l - 1; i++) {
    for (j = i + 1; j <= l; j++) {
      if (nums[i] + nums[j] === target) {
        return [i, j];
      }
    }
  }
}

//optimize solution using hashmap/object

 hashmap = {}
    for(let i =0; i<list.length; i++){
        n = list[i]
        rem = target - n;
        
        if(hashmap[rem] !== undefined){
            return [hashmap[rem], i]
        }
        else {
            hashmap[n] = i
        }
    }

console.log(twoSum([3, 7, 2, 2], 9));
